packages/allure-playwright/src/helpers.ts

```diff
@@ -0,0 +1,131 @@
+import { TestInfo } from "@playwright/test";
+import { Label, LabelName, Link, LinkType } from "allure-js-commons";
+
+
+export const ALLURE_METADATA_CONTENT_TYPE = "application/vnd.allure.metadata+json";
+
+
+export interface Metadata {
+  labels?: Label[];
+  links?: Link[];
+}
+
+export class AllurePlaywrightHelper {
+  private testInfo!: TestInfo;
+
+  constructor(testInfo: TestInfo) {
+    this.testInfo = testInfo;
+  }
+
+  addMetataAttachment(metadata: Metadata) {
+    this.testInfo.attachments.push({
```

Comment:
How about using a WeakMap for this data instead of writing a virtual file? (which could produce additional side-effects)

```
const testMetadata = new WeakMap();

addMetataAttachment() {
    let metadataStore = testMetadata.get(this.testInfo);
    if(!metadataStore) {
         metadataStore = [];
        testMetadata.set(this.testInfo);
    }
    metadata.push(metadata)
}
```